File: js/complete/no009.js
#!/usr/local/bin/node

/* There exists exactly one Pythagorean triplet for which
   a + b + c = 1000. Find the product abc. */

var timer = require('../timer.js'),
    operator = require('../operator.js');
3
function first_triplet(total) {
    var c;
    for(var a = 1; a < total - 1; a++) {
        for(var b = 1; b < total - a; b++) {
            c = total - a - b;
            if(a*a + b*b == c*c) {
                return [a, b, c];
            }
        }
    }

    return []; // None found
};

function main() {
    var triplet = first_triplet(1000),
        result = triplet.reduce(operator.mul, 1);
    return result;
};

timer.timer(9, main);

File: java/complete/no009.java
class no009 {
  public static int [] first_triplet(int total) {
    int c;
    for (int a = 1; a < total - 1; a++) {
      for (int b = 1; b < total - a; b++) {
        c = total - a - b;
        if (a * a + b * b == c * c) {
          return new int[] {a, b, c};
        }
      }
    }

    return new int[0];
  }

  public static int main(boolean verbose) {
    int [] triplet = first_triplet(1000);
    // In future, use reduce to multiply
    return triplet[0] * triplet[1] * triplet[2];
  }

  public static int main() {
    return main(false);
  }

  public static void main(String[] args) {
    System.out.println(main());
  }
}
